hypothesis-python/src/hypothesis/internal/conjecture/junkdrawer.py

```diff
@@ -140,9 +136,14 @@ def __ne__(self, other: object) -> bool:
         return self.__underlying != other.__underlying
 
     def append(self, n: int) -> None:
-        i = len(self)
-        self.__underlying.append(0)
-        self[i] = n
+        i = len(self.__underlying)
+        # try the fast path of appending n first. If this overflows, use the
+        # __setitem__ path, which will upgrade the underlying array.
+        try:
+            self.__underlying.append(n)
+        except OverflowError:
+            self.__underlying.append(0)
+            self[i] = n
```

Comment:
this is the only optimization in `junkdrawer.py`, the rest is code quality

Reply:
Could also compute `i =` in the OverflowError branch, for a tiny further speedup.